### PR TITLE
fix(cli): reuse the saved login session instead of re-opening browser OAuth

### PR DESCRIPTION
**Pull request by ItamarZand88**
## Summary

After one `alien login`, every later command opened the browser OAuth page again.
Release builds stored the tokens in the macOS keychain, and macOS ties a keychain item
to the code signature of the binary that created it — released CLI binaries are ad-hoc
signed, so after any upgrade the new binary was denied its own tokens. The login
bootstrap treated that, and every other failure, as "no session" and re-ran the browser
login. This keeps the session in a `credentials.json` file that survives upgrades, and
only opens the browser when there's no session left.

What happens when you run any platform command:

1. The CLI loads the saved session from `credentials.json` and uses it; an expired
   access token refreshes on its own.
2. If there's no usable session — nothing saved, or the login server rejected the
   refresh token — the browser login runs once, but only from a real terminal; an agent
   or script gets a `LOGIN_REQUIRED` error pointing at `alien login` instead of a
   browser that hangs.  ← the new behavior
3. Any other failure (a file it can't read, a network error during refresh) comes back
   as an error, not a browser.

`alien login` itself is unchanged — it always opens the browser (or prints the URL with
`--no-browser`), terminal or not.

Before: any failure re-opened the browser. After: the browser opens only when there's
no session; everything else comes back as an error you can read.

I also fixed `alien whoami`, which returned Unauthorized for every user session — the
API spec didn't declare the workspace the server needs to resolve a user, so the
generated client never sent it. The regenerated spec declares it and the CLI now sends
the workspace it already has.

I also hardened the login callback against two issues a review flagged on the
pre-existing callback code. First, the CSRF `state` check only ran when `state` was
present — a callback arriving with no `state` skipped the check and passed; `state` is a
guard independent of PKCE, so it's now mandatory (rejected unless present and matching).
Second, the success-redirect URL was written straight into an inline
`window.location.href = '...'` script; it's now escaped for that single-quoted-JS-string
context, so a quote, backslash, or `</script>` in it can't break out.

## How I tested it

Against api.alien.dev, logged in once:

- Ran `alien workspaces list` twice — silent both times, no browser. Rebuilt the release
  binary (new ad-hoc signature — the exact thing that used to break) and ran it again —
  still silent.
- Force-expired the saved access token — the next command refreshed it on its own.
  Replaced the refresh token with a junk value — the server's rejection came back as
  "Not logged in", exit 1, no browser.
- Ran `alien logout` — both files deleted, the next command failed right away with the
  structured error; restored the session and `alien whoami` returned my principal.
- Checked the agent case: `alien login` piped through a non-terminal still printed the
  authorize URL and waited — the terminal check only kicks in on the other commands, not
  on `alien login`.
- The token file is written 0600, and the CLI errors out instead of writing it to the
  working directory when it can't find a config dir, so it can't end up committed to a
  project. The refresh token only goes to the login server over a client with redirects
  turned off, so it can't be sent anywhere else.
- A forged login callback with a missing or wrong `state` is rejected — previously only a
  mismatched `state` was, so a callback carrying no `state` slipped through. Covered by
  `callback_state_must_be_present_and_match` in `crates/alien-cli/src/auth.rs`.
- The success-redirect URL is escaped before it's interpolated into the inline
  `window.location.href` script (`escape_js_single_quoted` in
  `crates/alien-cli/src/auth.rs`), so a `'`, `\`, or `</script>` in it can't break out of
  the JS string or the `<script>` element. Covered by
  `redirect_url_is_escaped_for_the_inline_script`.
- `cargo test -p alien-cli` → 96 pass: the 9 token-store tests (round-trip, the 0600
  mode, recovering from a corrupt file, refresh-token rotation, expiry edge cases), the
  callback-state test (missing, mismatched, and matching state), and the
  redirect-URL-escape test (breakout and line-terminator cases).

### CODE REVIEW

File: crates/alien-cli/src/auth.rs [hunk 1]
```diff
@@ -475,38 +480,20 @@ mod oauth_flow {
         true
     }
```
[the remark this answers is not in the document]
Comment (ItamarZand88) [reply crates/alien-cli/src/auth.rs:484]: Intentional. The provider issues JWTs with exp, so a token that won't parse is a corrupted store and a refresh repairs it. No reactive 401 refresh here, so erring toward a refresh beats sending a token we couldn't read. Added a comment.

**Comment by alongubkin** [comment]
Checked this against current `origin/main`.

The PR is no longer needed as a standalone change. The core behavior from this PR is already present on `main`:

- saved CLI session auth / `credentials.json` handling
- `LOGIN_REQUIRED` behavior for non-interactive auth
- OAuth callback state validation and success-redirect escaping
- `alien whoami` forwarding the workspace hint
- logout propagating deletion errors
- the `/v1/whoami` OpenAPI workspace parameter

The main auth-related files are byte-for-byte identical between `origin/main` and this PR head: `crates/alien-cli/src/auth.rs`, `crates/alien-cli/src/commands/whoami.rs`, `crates/alien-cli/src/error.rs`, and `crates/alien-cli/src/commands/platform/logout.rs`.

The remaining differences are newer `main` changes, especially follow-up workspace-context work and generated OpenAPI drift, not missing PR behavior. Closing this as superseded/already landed via later work.

**Comment by alongubkin** [comment]
Closing as superseded: the relevant CLI auth/session behavior has already landed on `main` via later work.
